src/main/java/mate/academy/spring/controller/ShoppingCartController.java

```diff
@@ -0,0 +1,52 @@
+package mate.academy.spring.controller;
+
+import mate.academy.spring.mapper.DtoResponseMapper;
+import mate.academy.spring.model.MovieSession;
+import mate.academy.spring.model.ShoppingCart;
+import mate.academy.spring.model.User;
+import mate.academy.spring.model.dto.response.ShoppingCartResponseDto;
+import mate.academy.spring.service.MovieSessionService;
+import mate.academy.spring.service.ShoppingCartService;
+import mate.academy.spring.service.UserService;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("shopping-carts")
+public class ShoppingCartController {
+    private final MovieSessionService movieSessionService;
+    private final UserService userService;
+    private final ShoppingCartService shoppingCartService;
+    private final DtoResponseMapper<ShoppingCartResponseDto, ShoppingCart> cartDtoResponseMapper;
+
+    public ShoppingCartController(UserService userService,
+                                  MovieSessionService movieSessionService,
+                                  ShoppingCartService shoppingCartService,
+                                  DtoResponseMapper<ShoppingCartResponseDto, ShoppingCart>
+                                          cartDtoResponseMapper) {
+        this.userService = userService;
+        this.movieSessionService = movieSessionService;
+        this.shoppingCartService = shoppingCartService;
+        this.cartDtoResponseMapper = cartDtoResponseMapper;
+    }
+
+    @PutMapping("/movie-sessions")
+    public ShoppingCartResponseDto addSession(@RequestParam Long userId,
+                                              @RequestParam Long movieSessionId) {
+        MovieSession movieSession = movieSessionService.get(movieSessionId);
+        User user = userService.get(userId);
+        shoppingCartService.addSession(movieSession, user);
+        ShoppingCart shoppingCart = shoppingCartService.getByUser(user);
+        return cartDtoResponseMapper.toDto(shoppingCart);
+    }
+
+    @GetMapping("/by-user")
```

Comment:
Similarly, the 'getByUser' method should have a more descriptive name, such as 'getShoppingCartByUserId'. This helps clarify the purpose of the method.